### PR TITLE
Feature/multi filter

### PR DESCRIPTION
**Pull request by chinyitan**

### CODE REVIEW

File: dolphin/processor/config.py [hunk 1]
```diff
@@ -157,15 +157,32 @@ def get_kwargs_model(self):
...
             'point_source_model_list': self.get_point_source_model_list(),
-            # 'index_lens_light_model_list': [[0]],
-            # 'index_source_light_model_list': [[0]],
+            'index_lens_light_model_list':
+                self.get_index_lens_light_model_list(),
+            'index_source_light_model_list':
+                self.get_index_source_light_model_list(),
         }
 
+        # For debugging multi-lensing
```
Comment (ajshajib) [inline dolphin/processor/config.py:172]: Remove these `print` statements.

File: dolphin/processor/config.py [hunk 2]
```diff
@@ -210,14 +228,28 @@ def get_kwargs_constraints(self):
...
+            'joint_lens_with_light': [],
+            'joint_lens_with_lens': []
         }
 
         if 'kwargs_constraints' in self.settings and self.settings[
                                             'kwargs_constraints'] is not None:
             for key, value in self.settings['kwargs_constraints'].items():
                 kwargs_constraints[key] = value
 
+        # For debugging multi-lensing
```
Comment (ajshajib) [inline dolphin/processor/config.py:240]: Remove these comment lines all together.

File: dolphin/processor/config.py [hunk 3]
```diff
@@ -411,7 +447,12 @@ def get_lens_light_model_list(self):
         :rtype:
         """
         if 'lens_light' in self.settings['model']:
-            return self.settings['model']['lens_light']
+            Combined_lens_light_model_list = []
```
Comment (ajshajib) [inline dolphin/processor/config.py:450]:
```suggestion
            combined_lens_light_model_list = []
```

File: dolphin/processor/config.py [hunk 4]
```diff
@@ -428,6 +469,44 @@ def get_point_source_model_list(self):
...
+    def get_index_lens_light_model_list(self):
+        """
+        Create list with of index for the different lens light profile
+         (for multiple filters)
+        """
+        if 'lens_light' in self.settings['model']:
+            index_lens_light_model_list = []
+            index_num = 0
+            for i in range(self.band_number):
+                Single_index_list = []
```
Comment (ajshajib) [inline dolphin/processor/config.py:481]:
```suggestion
                single_index_list = []
```

File: dolphin/processor/config.py [hunk 5]
```diff
@@ -564,6 +643,13 @@ def get_source_light_model_params(self):
         lower = []
         upper = []
 
+        for n in range(self.band_number-1):
+            self.settings['source_light_option']['n_max'].extend(
+                             self.settings['source_light_option']['n_max'])
+
+        # For debugging multi-lensing
```
Comment (ajshajib) [inline dolphin/processor/config.py:650]: Remove these commented out lines.

File: io_directory_example/settings/SL2SJ0219-0829_config.yml [hunk 6]
```diff
@@ -0,0 +1,47 @@
+---
+system_name: 'SL2SJ0219-0829'
```
Comment (ajshajib) [inline io_directory_example/settings/SL2SJ0219-0829_config.yml:2]: Change the SL2S naming for this, and use a generic name, e.g., "lens_system3". Change the data and psf files accordingly.

Comment (ajshajib) [reply io_directory_example/settings/SL2SJ0219-0829_config.yml:2]: If you are adding a third example system for checking multi-filter modeling, why did you add a second filter for "lens_system1"? Was that also necessary? If not, please remove unused files.

File: io_directory_example/settings/_test_config2.yml [hunk 7]
```diff
@@ -0,0 +1,56 @@
+# this config file is only used for testing, do not directly copy for use
```
Comment (ajshajib) [inline io_directory_example/settings/_test_config2.yml:1]: Is it possible to consolidate "_test_config1.yml" and "_test_config2.yml" into the corresponding *.yml file for "lens_system3"? The idea is one config file per system within the data folder. The io_directly_example directory is not only used for testing, but it is also an example for users on how to structure their data and config files.

File: io_directory_example/settings/lens_system3_config.yml [hunk 8]
```diff
@@ -0,0 +1,47 @@
+---
+system_name: 'SL2SJ0219-0829'
```
Comment (ajshajib) [inline io_directory_example/settings/lens_system3_config.yml:2]: `system_name` should say `'lens_system3'`.

Comment (chinyitan) [reply io_directory_example/settings/lens_system3_config.yml:2]: Thanks for spotting that

File: io_directory_example/settings/_test_config2.yml (hunk 7 above)
Comment (chinyitan) [reply io_directory_example/settings/_test_config2.yml:1]: I am renaming test_config2.yml to lens_system3.yml.  test_config1.yml has the same system as lens_system_1, and has been in the code before me, should I try to consolidate it?

**Comment by ajshajib** [comment]
I think, you can remove the notebook "Multi-filter example.ipynb", as it adds nothing totally new over "Basic example.ipynb". Probably you were using it to test the new codes, but within the package that purpose is served by the test functions. Also, please remove the associated output file for lens_system3.

File: io_directory_example/settings/_test_config2.yml (hunk 7 above)
Comment (ajshajib) [reply io_directory_example/settings/_test_config2.yml:1]: Ok, I see that, you can leave _test_config1.yml then, but please consolidate _test_config2.yml if possible. But, if it becomes too complex to do that, then you can leave that as a separate yml file too.

**Comment by chinyitan** [comment]
The initial reason why i kept the  "Multi-filter example.ipynb", was it shows how you can access the data for both filters.
 Maybe I can change the "Basic example.ipynb" to show that?

**Comment by ajshajib** [comment]
> The initial reason why i kept the "Multi-filter example.ipynb", was it shows how you can access the data for both filters.
> Maybe I can change the "Basic example.ipynb" to show that?

I don't think just that particular use case warrants a full example notebook. A user should be able to find the required argument for that feature from the documentation/docstring.

**Review by ajshajib — APPROVED** [review]

**Pull request merged**